queue_services/entity-filer/src/entity_filer/filing_processors/change_of_registration.py

```diff
@@ -99,89 +79,128 @@ def process(
-        party.email = party_info["officer"].get("email", "").lower()
-        party.identifier = party_info["officer"].get("identifier", "").upper()
-
-        # add addresses to party
-        if party_info.get("deliveryAddress", None):
-            if party.delivery_address:
-                update_address(
-                    party.delivery_address, party_info.get("deliveryAddress")
-                )
-            else:
-                address = create_address(
-                    party_info["deliveryAddress"], Address.DELIVERY
-                )
-                party.delivery_address = address
-        if party_info.get("mailingAddress", None):
-            if party.mailing_address:
-                update_address(party.mailing_address, party_info.get("mailingAddress"))
-            else:
-                mailing_address = create_address(
-                    party_info["mailingAddress"], Address.MAILING
-                )
-                party.mailing_address = mailing_address
-
-
-def _create_party_info(business, change_filing_rec, party_info):
-    party = merge_party(business_id=business.id, party_info=party_info, create=False)
-    for role_type in party_info.get("roles"):
-        role_str = role_type.get("roleType", "").lower()
-        role = {
-            "roleType": role_str,
-            "appointmentDate": role_type.get("appointmentDate", None),
-            "cessationDate": role_type.get("cessationDate", None),
-        }
-        party_role = create_role(party=party, role_info=role)
-        if party_role.role in [PartyRole.RoleTypes.COMPLETING_PARTY.value]:
-            change_filing_rec.filing_party_roles.append(party_role)
-        else:
-            business.party_roles.append(party_role)
-
-
 def post_process(business: LegalEntity, filing: Filing):
     """Post processing activities for change of registration.
 
     THIS SHOULD NOT ALTER THE MODEL
     """
     name_request.consume_nr(business, filing, "changeOfRegistration")
+
+def _update_partner_change(
+        legal_entity: LegalEntity,
+        change_filing_rec: Filing,
+        change_filing: Dict,
+        filing_meta: FilingMeta,
+):
+    name_request = dpath.util.get(change_filing, "/changeOfRegistration/nameRequest", default=None)
+    if name_request and (to_legal_name := name_request.get("legalName")):
+        alternate_name = AlternateName.find_by_identifier(legal_entity.identifier)
+        parties_dict = dpath.util.get(change_filing, "/changeOfRegistration/parties")
+
+        legal_entity.legal_name = get_partnership_name(parties_dict)
+
+        legal_entity.alternate_names.remove(alternate_name)
+        alternate_name.end_date = change_filing_rec.effective_date
+        alternate_name.change_filing_id = change_filing_rec.id
+        # alternate_name.delete()
+        db.session.add(alternate_name)
+        db.session.commit()
+        db.session.delete(alternate_name)
+        db.session.commit()
```

Comment:
can we remove `# alternate_name.delete()`?

Reply:
i kept it here to remind ourself to add delete function in business model and remove those 4 lines from here